app/controllers/api/base_controller/renderer.rb

```diff
@@ -75,10 +75,12 @@ def get_reftype(type, reftype, resource, _opts = {})
         return reftype unless resource.respond_to?(:attributes)
 
         rclass = resource.class
-        if collection_class(type) != rclass
-          matched_type = collection_config.name_for_klass(rclass)
-        end
-        matched_type || reftype
+        collection_class = collection_class(type)
+
+        # Ensures hrefs are consistent with those of the collection they were requested from
+        return reftype if collection_class.descendants.include?(rclass) || collection_class == rclass
```

Comment:
Hi @jntullo can you switch those 2 tests, collection_class == rclass would be more common, so we'd improve the performance here.